sphinxext/yamlinclude.py

```diff
@@ -0,0 +1,96 @@
+import os.path
+
+from docutils import statemachine
+
+from docutils.parsers.rst import directives
+
+from sphinx.util.docutils import SphinxDirective
+
+import yaml
+
+class YamlInclude(SphinxDirective):
+   
+   has_content = False
+   required_arguments = 1 # Must have the file to include
+   optional_arguments = 0 # Must only specify the file to include
+
+   option_spec = {
+      "key": directives.unchanged_required,
+      "subs" : directives.unchanged,
+      "includepath" : directives.unchanged,
+      "debug" : directives.flag,
+   }
+   
+   def run(self):
+      
+      # Subs should be specified as a multi-line indented set of comma-separated key values
+      # :subs:
+      #   foo,bar
+      #   bin,baz
+      #
+      # We parse that into a dictionary for lookup later as part of the substitution loop.
+      subs = {}
+      if ('subs') in self.options:
+         for i in self.options.get('subs').splitlines():
+            kv = i.split(',')
+            subs[kv[0]] = kv[1]
+
+      if ('debug') in self.options: print ("Substitution dictionary is "); print(subs)
+
+      key = self.options.get('key')
```

Comment:
You probably want `self.options["key"]` unless you actually want to handle the case where `key` doesn't exist